ui/src/app/layout/createtopic/datasettings/data-settings.component.ts

```diff
@@ -0,0 +1,177 @@
+
+    selectedCleanUpStrategy: CleanUpStrategy[] = ['delete'];
+
+    compactionTime = 24;
+
+    compactionTimeUnit = 'h';
+
+    retentionTime = 7;
+
+    retentionTimeUnit = 'd';
+
+    criticalityType: Criticality = 'NORMAL';
+
+    dataIntervals: string[] = ['< 1.000', '< 100.000', '< 1.000.000', '> 1.000.000'];
+
+    selectedDataSliderValue: MessagesPerDay;
+
+    sizeIntervals: string[] = ['< 1 KB', '< 10 KB', '< 100 KB', '< 1 MB', '> 1 MB'];
+
+    selectedSizeSliderValue;
+
+    optionsData: Options = {
+        stepsArray: [
+            { value: 0 },
+            { value: 1 },
+            { value: 2 },
+            { value: 3 },
+            { value: 4 }
+        ],
+        step: 1,
+        showTicks: true,
+        showTicksValues: true,
+        translate: (value: number): string => this.dataIntervals[value - 1]
+    };
+
+    optionsSize: Options = {
+        stepsArray: [
+            { value: 0 },
+            { value: 1 },
+            { value: 2 },
+            { value: 3 },
+            { value: 4 },
+            { value: 5 }
+        ],
+        step: 1,
+        showTicks: true,
+        showTicksValues: true,
+
+        translate: (value: number): string => this.sizeIntervals[value - 1]
+    };
+
+    constructor() {
+    }
+
+    deleteItem(deletion: string) {
+
+        this.selectedCleanUpStrategy = this.selectedCleanUpStrategy.filter(strategy => strategy !== deletion);
+    }
+
+    handleSubscriptionApprovalRequiredChange(): void {
+
+        if (this.selectedCleanUpStrategy.length === 1 && this.selectedCleanUpStrategy.includes('delete')) {
+            return;
+        }
+
+        if (this.subscriptionApprovalRequired && !this.selectedCleanUpStrategy.includes('delete')) {
+            this.selectedCleanUpStrategy.push('delete');
+        }
+
+    }
+
+    prepareDataForParent(): Promise<any> {
+        const initialSettings: TopicSettingsData = {
+            subscriptionApprovalRequired: this.subscriptionApprovalRequired,
+            cleanUpStrategy: this.selectedCleanUpStrategy,
+            compactionTimeMillis: this.toMilliSeconds(Number(this.compactionTime), this.compactionTimeUnit),
+            retentionTimeMillis: this.toMilliSeconds(Number(this.retentionTime), this.retentionTimeUnit),
+            criticality: this.criticalityType,
+            messagesPerDay: this.selectedDataSliderValue,
+            messagesSize: this.selectedSizeSliderValue
+        };
+        return Promise.resolve(this.setTopicSettings.emit(initialSettings));
+    }
+
+    onUserChangeEndData(changeContext: ChangeContext) {
+
+        this.selectedDataSliderValue = this.resolveData(changeContext.value);
+    }
+
+    onUserChangeEndSize(changeContext: ChangeContext) {
+
+        this.selectedSizeSliderValue = this.resolveSize(changeContext.value);
+    }
+
+    private toMilliSeconds(time, unit) {
+        return moment.duration(time, unit).asMilliseconds();
+    }
+
+    private resolveData(index: number): MessagesPerDay {
```

Comment:
Can't this be encoded in the options of the slider (where it is now `{ value: 1}`)?